### PR TITLE
delete empty fastqs

### PR DESCRIPTION
**Pull request by moonso**
This PR adds a functionality so that empty fastq files will be removed from housekeeper and from disk. Atm there are several cases where there are fastq files added to HK even though they have no content. While looping over the cases to check which cases that are ready for fastq compression the compress API will remove the empty fastq files from housekeeper and from disk.

### How to test
Tests can only be run in automated test suite

## Review
- [ ] code approved by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/models/file_data.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        handle = open(path, "rb")
+        if handle.read(2).hex() == GZIP_MAGIC_NUMBER:
+            return True
+        return False
+
+    @staticmethod
+    def is_empty(path: Path) -> bool:
+        """Check if file is empty"""
+        handle = open(path, "rb")
+        if handle.read(1):
```
Comment (henrikstranneheim) [inline cg/models/file_data.py:25]: What does "1" stand for? Could you give it a named constant to make it easier to understand?

File: tests/conftest.py [hunk 2]
```diff
@@ -235,16 +241,27 @@ def fixture_run_name() -> str:
     return "fastq_run"
 
 
+@pytest.fixture(scope="function", name="original_fastqs")
+def fixture_original_fastq(fastq_dir: Path, run_name) -> CompressionData:
```
Comment (henrikstranneheim) [inline tests/conftest.py:245]:
```suggestion
def fixture_original_fastq_path(fastq_dir: Path, run_name) -> CompressionData:
```

File: tests/models/conftest.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        " ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ull"
+        "amco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehende"
+        "rit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaec"
+        "at cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
+    )
+    return _content
+
+
+@pytest.fixture(name="file_path")
+def fixture_file_path(project_dir: Path) -> Path:
```
Comment (henrikstranneheim) [inline tests/models/conftest.py:23]: I think this name should better reflect the docstring

File: tests/models/conftest.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    """Return the path to a non existing file"""
+    return project_dir / "a_file.txt"
+
+
+@pytest.fixture(name="gzipped_file_path")
+def fixture_gzipped_file_path(file_path: Path) -> Path:
```
Comment (henrikstranneheim) [inline tests/models/conftest.py:29]: Same here: I would expect to get a path to an existing file

File: tests/models/test_file_data.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    # WHEN checking if the file is gzipped
+    res = FileData.is_gzipped(filled_file)
+
+    # THEN assert the result is False
+    assert res is False
+
+
+def test_is_gzipped_non_existing(gzipped_file_path):
+    """Test if a file that is does not exist is gzipped"""
```
Comment (henrikstranneheim) [inline tests/models/test_file_data.py:36]:
```suggestion
    """Test if a file that does not exist is gzipped"""
```

File: tests/models/test_file_data.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    assert not gzipped_file_path.exists()
+
+    # WHEN checking if the file is gzipped
+    with pytest.raises(FileNotFoundError):
+        # THEN assert that a FileNotFoundError is raised
+        FileData.is_gzipped(gzipped_file_path)
+
+
+def test_is_gzipped_empty_file(gzipped_file_path):
+    """Test if a file that is does exists but is empty"""
```
Comment (henrikstranneheim) [inline tests/models/test_file_data.py:47]:
```suggestion
    """Test if a file that does exists but is empty"""
```

**Review by henrikstranneheim — COMMENTED** [review]
It looks good! Just to be certain - this is deleting fastq (presumed empty). The repercussion if there is something wrong the empty fastq file check could be quite grave. Do we dare to this automatically? Would it be possible to add a couple of cases to test this on alongside the unit tests

File: cg/models/file_data.py (hunk 1 above)
Comment (moonso) [reply cg/models/file_data.py:25]: It is just reading the first byte of the file. I can make a comment in the code

File: tests/conftest.py (hunk 2 above)
Comment (moonso) [reply tests/conftest.py:245]: It is not returning a path but a CompressionData object so the path suffix is not in place here

Comment (moonso) [reply tests/conftest.py:245]: Changed to `original_fastq_data`

File: tests/models/conftest.py (hunk 3 above)
Comment (moonso) [reply tests/models/conftest.py:23]: changed to `non_existing_gzipped_file_path`

File: tests/models/conftest.py (hunk 4 above)
Comment (moonso) [reply tests/models/conftest.py:29]: Fixed

**Comment by moonso** [comment]
Skärmavbild 2020-09-01 kl  17 39 50

Merged and deployed on hasta

**Pull request merged**